includes/components/class-attribute.php

```diff
@@ -1130,6 +1187,11 @@ public function add_sort_options( $form_args, $form ) {
 		// Set options.
 		$form_args['fields']['_sort']['options'] = array_merge( $form_args['fields']['_sort']['options'], $options );
 
+		// Set default sort option.
+		if ( get_option( hp\prefix( $model . '_sorting_option' ) ) ) {
+			$form_args['fields']['_sort']['default'] = get_option( hp\prefix( $model . '_sorting_option' ) );
```

Comment:
в листинг сорт форме значения меняется в зависимости от настройки, но сортировка не работает на фронте. если есть вопросы по реализации или в тупик зашел, то пиши сразу

Reply:
да, значение сортировки изменяется пока только визуально

пробовал делать изменения в хуках `pre_get_posts` и `hivepress/v1/forms/model_sort` - не срабатывает

Reply:
попробуй хук `posts_orderby`. проверки страницы можешь ориентироваться на роуты `listings_view_page`, `vendors_view_page` и `requests_view_page`, но на практике глянь какие условия нужны, чтоб сортировка работала только на страницах, где отображаются, например, все листинги

Reply:
This one should've worked, if setting the default value doesn't work for this code https://github.com/hivepress/hivepress/blob/master/includes/components/class-attribute.php#L1761 to fetch the correct option by default, then maybe also moving this option to the top will work? Anyway, it's better to research this further and check why the default value is ok on the front-end but it's not fetched on the linked code line correctly.

Once fetched correctly (e.g. when you open the Listings page without performing any search) then no additional changes may be required because the sorting logic is already there.